client/src/panels/decompositionPanel.js

```diff
@@ -104,6 +105,32 @@ class DecompositionPanel extends React.Component {
     this.updateDataModel('' + result.maxPersistenceLevel);
   }
 
+  componentDidMount() {
+    if (this.state.decompositionMode == 'Morse-Smale') {
+      let k = 15; // num nearest neighbors to consider when generating M-S complex for a dataset
+      let datasetId = this.props.dataset.datasetId;
```

Comment:
datasetId added to this.state. Why does this function still uses this.props' value? @kylinmb

Reply:
related: updateDataModel and handeDecompositionModeChange also use this.props version.

Reply:
datasetId added to this.state. Why does this function still uses this.props' value? This should be changed to this.state.datasetId, I updated it on line 94 but missed it in these places.